### ISSUE TITLE
Obtain wrong IP address when incoming comments arrive in Kubernetes environment

### ISSUE THREAD

**Issue by JohnNiang** [halo-dev/halo#4054]
### What is version of Halo has the issue?

2.7.0-SNAPSHOT

### What database are you using?

PostgreSQL

### What is your deployment method?

Docker

### What happened?

I built my Halo in Kubernetes environment, which is using Traefik as my Ingress Controller. But I found that all IP address in comment list were internal IP address `10.42.0.1`.

```diff
{
  "spec": {
    "raw": "666",
    "content": "666",
    "owner": {
      "kind": "Email",
      "name": "well-wisher@halowrite.com",
      "displayName": "Well Wisher",
      "annotations": {
        "website": "https://halowrite.com/"
      }
    },
    "userAgent": "Mozilla/5.0 (iPhone; CPU iPhone OS 16_1_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 MicroMessenger/8.0.34(0x18002234) NetType/WIFI Language/en",
+   "ipAddress": "10.42.0.1",
    "approvedTime": "2023-04-27T14:35:45.208137429Z",
    "creationTime": "2023-04-27T14:35:45.208138680Z",
    "priority": 0,
    "top": false,
    "allowNotification": true,
    "approved": true,
    "hidden": false,
    "subjectRef": {
      "group": "content.halo.run",
      "version": "v1alpha1",
      "kind": "SinglePage",
      "name": "0200a1da-df7e-4f9c-bd55-fddc15ebb218"
    }
  },
  "status": {
    "unreadReplyCount": 0,
    "hasNewReply": false
  },
  "apiVersion": "content.halo.run/v1alpha1",
  "kind": "Comment",
  "metadata": {
    "finalizers": [
      "comment-protection"
    ],
    "name": "5cf84ad6-a38c-43fe-88e6-e602c8b9c2a2",
    "annotations": {},
    "version": 2,
    "creationTimestamp": "2023-04-27T14:35:45.208224772Z"
  }
}
```

Please feel free to correct me if I overlooked something.

### Additional information

/kind bug
/area core

**Comment by JohnNiang** [comment]
After setting up echo server, I got the following result:

```bash
curl https://echo.halowrite.com
```
```bash
Hostname: whoami-844bdd7c79-7h24j
IP: 127.0.0.1
IP: ::1
IP: 10.42.0.56
IP: fe80::1875:45ff:feef:d9c3
RemoteAddr: 10.42.0.11:42198
GET / HTTP/1.1
Host: echo.halowrite.com
User-Agent: curl/8.0.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 10.42.0.1
X-Forwarded-Host: echo.halowrite.com
X-Forwarded-Port: 443
X-Forwarded-Proto: https
X-Forwarded-Server: traefik-bb69b68cd-rgc2p
X-Real-Ip: 10.42.0.1
```

**Comment by JohnNiang** [comment]
I found a solution to fix the problem if you deployed Halo in K3s:

```bash
[root@main-node manifests]# cat traefik-config.yaml
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: traefik
  namespace: kube-system
spec:
  valuesContent: |-
    service:
      spec:
        externalTrafficPolicy: "Local"

[root@main-node manifests]# pwd
/var/lib/rancher/k3s/server/manifests
```

**Comment by guqing** [comment]
Fixed by #4139
